masq_lib/src/percentage.rs

```diff
@@ -80,178 +229,418 @@ impl Percentage {
-                    num,
-                )
-            })
-            .for_each(|(panic_msg, num)| {
+            .map(|num| (PurePercentage::try_from(num), num))
+            .for_each(|(res, num)| {
                 assert_eq!(
-                    panic_msg,
-                    format!("Accepts only range from 0 to 100 but {} was supplied", num)
+                    res,
+                    Err(format!(
+                        "Accepts only range from 0 to 100 but {} was supplied",
+                        num
+                    ))
                 )
             });
     }
 
+    struct Case {
+        requested_percent: u32,
+        examined_base_number: i64,
+        expected_result: i64,
+    }
+
     #[test]
     fn too_low_values() {
-        vec![((10, 1), 0), ((9, 1), 0), ((5, 14), 1), ((55, 40), 22)]
-            .into_iter()
-            .for_each(|((per_cent, examined_number), expected_result)| {
-                let result = Percentage::new(per_cent).of(examined_number);
-                assert_eq!(
-                    result, expected_result,
-                    "For {} per cent and number {} the expected result was {} but we got {}",
-                    per_cent, examined_number, expected_result, result
-                )
-            })
+        vec![
+            Case {
+                requested_percent: 49,
+                examined_base_number: 1,
+                expected_result: 0,
+            },
+            Case {
+                requested_percent: 9,
+                examined_base_number: 1,
+                expected_result: 0,
+            },
+            Case {
+                requested_percent: 5,
+                examined_base_number: 14,
+                expected_result: 1,
+            },
+            Case {
+                requested_percent: 55,
+                examined_base_number: 41,
+                expected_result: 23,
+            },
+            Case {
+                requested_percent: 55,
+                examined_base_number: 40,
+                expected_result: 22,
+            },
+        ]
+        .into_iter()
+        .for_each(|case| {
+            let result = PurePercentage::try_from(u8::try_from(case.requested_percent).unwrap())
+                .unwrap()
+                .of(case.examined_base_number);
+            assert_eq!(
+                result, case.expected_result,
+                "For {} percent and number {} the expected result was {} but we got {}",
+                case.requested_percent, case.examined_base_number, case.expected_result, result
+            )
+        })
     }
 
     #[test]
-    fn should_be_rounded_down_works_for_last_but_one_digit() {
+    fn should_be_rounded_to_works_for_last_but_one_digit() {
         [
-            (787879, false),
-            (1114545, true),
-            (100, true),
-            (49, true),
-            (50, false),
+            (49, RoundingTo::SmallerPositive, RoundingTo::SmallerNegative),
```

Comment:
As we both were looking at this test, an idea came up in your mind, to call these enums as `BelowFifty` and `AboveFifty` in place of `Smaller` and `Bigger`.

It sounds like a good improvement to me. 👍